### PR TITLE
feat(workflow): Implementing metric alert rule signal + receiver to track creation

### PR DESCRIPTION
**Pull request by iProgramStuff**
This implements a signal and receiver to track metric alert creation and feature adoption.

TODO(?): Should I track the actions / thresholds as well?

Addresses https://app.asana.com/0/1143846759074121/1173082414526457

### CODE REVIEW

File: src/sentry/incidents/endpoints/project_alert_rule_index.py [hunk 1]
```diff
@@ -68,6 +69,9 @@ def post(self, request, project):
 
         if serializer.is_valid():
             alert_rule = serializer.save()
+            metric_alert_rule_created.send_robust(
```
Comment (wedamija) [inline src/sentry/incidents/endpoints/project_alert_rule_index.py:72]: Why do we fire this here? Would it make sense to just use `post_save` on `AlertRule`?

File: src/sentry/models/featureadoption.py [hunk 2]
```diff
@@ -73,6 +73,7 @@
 manager.add(65, "saved_search", "Saved Search", "web", prerequisite=["advanced_search"])
 manager.add(66, "inbound_filters", "Inbound Filters", "web", prerequisite=["first_event"])
 manager.add(67, "alert_rules", "Alert Rules", "web", prerequisite=["first_event"])
+manager.add(67, "metric_alert_rules", "Metric Alert Rules", "web", prerequisite=["first_event"])
```
[the remark this answers is not in the document]
Comment (iProgramStuff) [reply src/sentry/models/featureadoption.py]: Probably.

File: src/sentry/incidents/endpoints/project_alert_rule_index.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/incidents/endpoints/project_alert_rule_index.py:72]: Would the user be available there? We don't store it on the rule.

File: src/sentry/incidents/endpoints/project_alert_rule_index.py [hunk 3]
```diff
@@ -68,6 +69,9 @@ def post(self, request, project):
 
         if serializer.is_valid():
```
Comment (EvanPurkhiser) [inline src/sentry/incidents/endpoints/project_alert_rule_index.py:70]: IMO reverse this if case to be

```py
if not serializer.is_valid():
  return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)

# do valid things
```

File: src/sentry/incidents/endpoints/project_alert_rule_index.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/incidents/endpoints/project_alert_rule_index.py:72]: Ahh, I missed that... You can use 
```
from sentry.app import env
env.request
```

`env` stores the request in a global place. You'll need to check that the `request` isn't `None` in the signal handler.

Comment (EvanPurkhiser) [reply src/sentry/incidents/endpoints/project_alert_rule_index.py:72]: I'm personally in favor of just keeping it simple like this. I actually didn't know about `env.request`

**Review by wedamija — APPROVED** [review]

**Pull request merged**